### ISSUE TITLE
Keycloak 18. Cannot to connect to Postgres during start

### ISSUE THREAD

**Issue by alexeytopolnitskiy** [keycloak/keycloak#11797]
### Describe the bug

I am following the documentation to setup Keycloak 18 in production mode on bare metal.
I set up hostname, SSL/TLS certificates and DB data in `conf/keycloak.conf`.  Also exported admin user credentials via environment variables.
---
```
db=postgres
db-username=user
db-password=password
db-url-host=localhost
db-url-port=5432
db-url-database=keycloak
```
---
When I run `bin/kc.sh build` and then `bin/kc.sh start` it returns me:
```
2022-05-02 16:59:00,082 INFO  [org.keycloak.quarkus.runtime.hostname.DefaultHostnameProvider] (main) Hostname settings: FrontEnd: *my_hostname*, Strict HTTPS: true, Path: <request>, Strict BackChannel: false, Admin: <request>, Port: -1, Proxied: true
2022-05-02 16:56:36,297 WARN  [io.agroal.pool] (agroal-11) Datasource '<default>': Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
2022-05-02 16:56:36,302 WARN  [org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator] (JPA Startup Thread: keycloak-default) HHH000342: Could not obtain connection to query metadata: org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
2022-05-02 16:59:07,974 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (production) mode
2022-05-02 16:59:07,975 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to obtain JDBC connection
```

5432 port is open on the host. Also I tried different `db-url-host` (hostname, url, 127.0.0.1, IP) but always received the same result nothing helped.

### Version

18.0

### Expected behavior

Keycloak connects to Postgres and starts

### Actual behavior

Fails to connect to database during `bin/kc.sh start` command

**Comment by DGuhr** [comment]
Hi @alexeytopolnitskiy 

I can not reproduce your issue, I tried it just now using the config you see in this screenshot and freshly downloaded kc18 dist, and it worked like a charm:
<img width="1792" alt="Screenshot 2022-05-02 at 20 20 56" src="https://user-images.githubusercontent.com/89905860/166303504-30a50996-1777-4697-b17c-274b4dfe5929.png">

Only difference is I spun up the db using a docker container, but nevertheless the db is reachable on localhost:5432 as i forward the port, so it is essentially the same. That said, this is most likely a problem with your environment on your machine. 

Maybe some firewall or sth in that area that prevents calls to localhost on some / all ports?

Nevertheless, sorry but i'll close here, as this is not a bug in Keycloak itself. Feel free to re-open when I missed something :)

**Comment by alexeytopolnitskiy** [comment]
@DGuhr Thank you for your reply!

I just do not understand one thing that I did not find in docs. When I set keycloak up in bare metal (without containers and k8s), do I have to set postgres up my own (because that's what You did as I understand)? And what exactly happens when I run `bin/kc.sh build`? Does this command really start Postgres? And how this "builds" are saved, maybe I want to return to some previous builds?

Also, maybe I missed something when I was preparing the cluster. I installed Java JDK, made certificates. Maybe there is something else that must be installed?
